### PR TITLE
GAWB-487: when loading workflow, don't load the entire submission

### PR DESCRIPTION
**Pull request by davidangb**
Previously, when navigating through the UI to the details of a single workflow, we would load the entire submission. If that submission contained a lot of other workflows, this was a very heavyweight operation just to get a single workflow.

Note to reviewer: a lot of the diff below is whitespace; try with ?w=1 in the github url to ignore whitespace.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala [hunk 1]
```diff
@@ -223,6 +227,10 @@ trait WorkflowComponent {
       filter(_.id === id)
     }
 
+    def findWorkflowByExternalIdAndSubmissionId(externalId: String, submissionId: UUID): WorkflowQueryType = {
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala:230]: I see in the schema that the workflow id is called external_id (not sure why), but it seems like this function and getByExternalId would be clearer to read if the word External was replaced with Workflow

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -1117,22 +1117,20 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: SlickDataSo
...
-              case (Failure(outputsFailure), Success(logs)) =>
-                throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadGateway, s"Unable to get outputs for ${submissionId}.", executionServiceDAO.toErrorReport(outputsFailure)))
-              case (Success(outputs), Failure(logsFailure)) =>
-                throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadGateway, s"Unable to get logs for ${submissionId}.", executionServiceDAO.toErrorReport(logsFailure)))
-              case (Failure(outputsFailure), Failure(logsFailure)) =>
-                throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadGateway, s"Unable to get outputs and unable to get logs for ${submissionId}.",
-                  Seq(executionServiceDAO.toErrorReport(outputsFailure),executionServiceDAO.toErrorReport(logsFailure))))
-            })
-          }
+        withWorkflow(workspaceName, submissionId, workflowId, dataAccess) { workflow =>
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:1120]: seems like a huge improvement, great!

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala (hunk 1 above)
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala:230]: The primary key of WORKFLOW is ID, a bigint - which is what I think of when I hear "workflowId". The external id (WORKFLOW.EXTERNAL_ID) is a UUID that is visible in the UI.

Within the code, a lot of places use integer ID ... but this particular use case relies on the UUID, so I am explicitly naming it "externalId"). Does that help?

Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala:230]: yeah that's reasonable

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala:230]: we used to use cromwell's workflow id as our primary key but that caused unit test problems and is a bad design for the future so there is now an external id which is cromwell's workflow id and our own numeric PK

**Comment by dvoet** [comment]
:+1:

**Comment by abaumann** [comment]
:+1:

**Pull request merged**